source/MRMesh/MRColorMapAggregator.cpp

```diff
@@ -0,0 +1,183 @@
+#include "MRColorMapAggregator.h"
+#include "MRGTest.h"
+
+namespace MR
+{
+
+template<typename Tag>
+int lastValid( TaggedBitSet<Tag> elementBitSet )
+{
+    if ( !elementBitSet.any() )
+        return -1;
+    for ( int i = int( elementBitSet.size() ) - 1; i >= 0; --i )
+    {
+        if ( elementBitSet.test( Id<Tag>( i ) ) )
+            return i;
+    }
+    return -1;
+}
```

Comment:
I think this should me method of MR::BitSet, but it is work for another PR

Reply:
reworked